// @framer/ea.quantum/code/Buttons/ButtonHollowText.tsx
import * as React from 'react';
import { PropertyControls, ControlType } from 'framer';
import styled, { css } from 'styled-components';
import * as tokens from '../tokens';

// Define type of property
interface Props {
    text: string;
    size: string;
    isEnabled: boolean;
    width: number;
    isFullWidth: string;
    height: number;
    onClick(e: any): void;
}

const ButtonHT = styled.button`
    position: relative;
    box-sizing: border-box;
    background: transparent;
    align-items: center;
    border: none;
    padding: 0;
    margin: 0;
    display: inline-flex;
    justify-content: center;
    .buttonInner {
      display: flex;
      &:after {
        content: '';
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
        position: absolute;
        border-radius: 16px;
        box-sizing: border-box;
        z-index: -1;
      }
    }
    span {
      font-family: "BrownPro", "Helvetica Neue", "Arial", sans-serif;
      font-weight: bold;
      line-height: 16px;
      letter-spacing: 2px;
      text-transform: uppercase;
      display: block;
      color: #FFF;
      text-align: center;
      white-space: nowrap;
    }

    ${props =>
        props.isEnabled == true &&
        css`
            .buttonInner {
                &:after {
                    background: ${tokens.ColorButton.hollow};
                    border: solid 2px ${tokens.ColorBrand.primary};
                    transition: all 125ms cubic-bezier(0.77, 0, 0.175, 1);
                }
            }

            &:hover {
                .buttonInner {
                    &:after {
                        transform: scale(0.96);
                        border: solid 2px #2d60cb;
                        background: rgb(23, 26, 41, 0.8);
                    }
                }
            }

            &:active {
                .buttonInner {
                    &:after {
                        transform: scale(0.96);
                        border: solid 2px #427ffe;
                        background: rgb(23, 26, 41, 0.6);
                    }
                }
            }

            &:focus {
                .buttonInner {
                    &:after {
                        border: solid 2px #ffffff;
                        background: rgb(23, 26, 41, 0.6);
                    }
                }
            }
        `}

    ${props =>
        props.isEnabled == false &&
        css`
            cursor: not-allowed;
            .buttonInner {
                span {
                    color: ${tokens.ColorNeutral._300};
                }
                &:after {
                    background: ${tokens.ColorButton.ghost};
                    border: 2px solid ${tokens.ColorNeutral._400};
                }
            }
        `}

    ${props =>
        props.size == 'small' &&
        css`
            height: 32px;
            min-width: 76px;
            max-width: 160px;
            .buttonInner {
                padding: ${tokens.PaddingButtonSquishDefault.small};
                &:after {
                    border-radius: ${tokens.BorderRadiusDefault.smaller};
                }
            }
            span {
                font-size: ${tokens.TextSizeButtonDefault.small};
            }
        `}

    ${props =>
        props.size == 'medium' &&
        css`
            height: 48px;
            min-width: 112px;
            max-width: 240px;
            .buttonInner {
                padding: ${tokens.PaddingButtonSquishDefault.medium};
                &:after {
                    border-radius: ${tokens.BorderRadiusDefault.smaller};
                }
            }
            span {
                font-size: ${tokens.TextSizeButtonDefault.medium};
            }
        `}

    ${props =>
        props.size == 'large' &&
        css`
            height: 64px;
            min-width: 132px;
            max-width: 256px;
            .buttonInner {
                padding: ${tokens.PaddingButtonSquishDefault.large};
                &:after {
                    border-radius: ${tokens.BorderRadiusDefault.medium};
                }
            }
            span {
                font-size: ${tokens.TextSizeButtonDefault.large};
            }
        `}
    ${props =>
        props.isFullWidth == 'fullWidth' &&
        css`
            width: 100%;
            min-width: 100%;
            max-width: 100%;
        `}
`;

export class ButtonHollowText extends React.Component<Props> {
    // Set default properties
    static defaultProps = {
        text: 'Label',
        size: 'small',
        isEnabled: true,
        width: 76,
        height: 32,
        onClick: () => null
    };
    // Items shown in property panel
    static propertyControls: PropertyControls = {
        text: {
            type: ControlType.String,
            title: 'Label'
        },
        size: {
            type: ControlType.Enum,
            title: 'Size',
            options: ['small', 'medium', 'large'],
            optionTitles: ['Small', 'Medium', 'Large']
        },
        isEnabled: {
            type: ControlType.Boolean,
            title: 'Enabled'
        },
        isFullWidth: {
            type: ControlType.SegmentedEnum,
            title: 'Button Width',
            options: ['standard', 'fullWidth'],
            optionTitles: ['Standard', 'Full Width']
        }
    };

    render() {
        return (
            <ButtonHT
                isEnabled={this.props.isEnabled}
                size={this.props.size}
                onClick={this.props.onClick}
                isFullWidth={this.props.isFullWidth}>
                <div className='buttonInner'>
                    <span>{this.props.text}</span>
                </div>
            </ButtonHT>
        );
    }
}
