// @framer/ea.quantum/code/AccordionContainer.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import styled, { css } from "styled-components";

// Define type of property
interface Props {
  state: string;
  width: number;
  height: number;
}

const AccordionContainerTest = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  position: static !important;
  height: auto !important;
  width: 100%;
  overflow-y: visible;

  & > * {
    position: static !important;
    height: 0px !important;
    display: flex !important;
    flex-direction: column;

    & > * {
      overflow: visible !important;
      position: static !important;

      & > * {
        position: static !important;
        height: auto !important;
        transform: initial !important;
        width: 100% !important;
      }
    }
  }
`;

export class AccordionContainer extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    state: "default",
    titile: "Title",
    width: 240,
    height: 300
  };
  // Items shown in property panel
  static propertyControls: PropertyControls = {};

  state = { toggle: true };

  render() {
    return (
      <AccordionContainerTest state={this.props.state}>
        {this.props.children}
      </AccordionContainerTest>
    );
  }
}
