// @framer/ea.quantum/code/AvatarNoIndicator.tsx
import * as React from 'react';
import { PropertyControls, ControlType } from 'framer';
import styled from 'styled-components';
import * as tokens from './tokens';

const AvatarNoDot = styled.div`
    height: ${props => parseFloat(props.size)}px;
    width: ${props => parseFloat(props.size)}px;
    position: relative;
    background: url(${props => props.image});
    background-color: gray;
    background-size: cover;
    background-position: center center;
    background-repeat: no-repeat;
    transition: transform ${tokens.AnimationTimeDefault.average};
    /* border: 5px solid transparent; */

    &:hover {
        transform: scale(1.1);
        transition-timing-function: ${tokens.AnimationEasing.standard};
        box-shadow: inset 0 0 0 2px white;
    }
`;

// Define type of property
interface Props {
    image: string;
    size: string;
}

export class AvatarNoIndicator extends React.Component<Props> {
    // Set default properties
    static defaultProps = {
        size: '48',
        width: 48,
        height: 48
    };

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        image: { type: ControlType.Image, title: 'Image' },
        size: {
            type: ControlType.Enum,
            title: 'Size',
            options: ['16', '24', '32', '48', '64', '96'],
            optionTitles: ['Smallest', 'Small', 'Medium', 'Large', 'Larger', 'Huge']
        }
    };

    render() {
        return (
            <AvatarNoDot
                size={this.props.size}
                image={this.props.image}
                style={{
                    borderRadius:
                        this.props.size === '16' ? 4 : this.props.size === '24' ? 8 : this.props.size === '32' ? 10 : 12
                }}
            />
        );
    }
}
