// @framer/ea.quantum/code/Avatar.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { QuantumComponent, TokenConsumer } from "./QuantumComponent";
import styled, { css } from "styled-components";

const AvatarContainer = styled.div`
  position: relative;
  height: 100%;
`;

// Define type of property
interface Props {
  size: string;
  state: string;
  width: number;
  height: number;
  image: string;
  style?: any;
}

// const style = tokens => styled.div`
//   ${props => props.size == "default" && css``}
// `;

export class Avatar extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    size: "51",
    state: "rgb(0, 255, 132)",
    width: 48,
    height: 48
  };
  // Items shown in property panel
  static propertyControls: PropertyControls = {
    size: {
      type: ControlType.Enum,
      title: "Size",
      options: ["16", "26", "34", "51", "68", "103"],
      optionTitles: ["Smallest", "Small", "Medium", "Large", "Larger", "Huge"]
    },
    state: {
      type: ControlType.SegmentedEnum,
      title: "State",
      options: ["rgb(0, 255, 132)", "yellow", "red", "gray"],
      optionTitles: ["Online", "Away", "Busy", "Offline"]
    },

    image: {
      type: ControlType.Image,
      title: "User Image"
    }
  };

  render() {
    return (
      //   <QuantumComponent>
      //     <TokenConsumer>
      //       {tokens => {
      //         const Avatar = style(tokens);
      //         return (
      <AvatarContainer style={...this.props.style}>
        <div>
          <svg
            width={parseFloat(this.props.size)}
            height={parseFloat(this.props.size)}
            viewBox="0 0 50 51"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g id="Symbols" fill="#fff" fillRule="evenodd">
              <mask id="mask">
                <path
                  d="M45.6576325,35.1186647 C45.1970411,35.0395549 44.7278302,35 44.25,35 C39.6936508,35 36.1614463,38.696512 36,43.25 C35.9731743,44.0066003 36.0551254,44.7360076 36.2458532,45.4382219 L36.2458558,45.4382212 C36.5353781,46.5041733 35.9059574,47.6030024 34.8400053,47.8925247 C34.6956392,47.9317359 34.5473295,47.9546259 34.3978591,47.9607654 C33.7610579,47.9869218 33.1673345,48 32.6166888,48 L15.3833112,48 C10.0341988,48 8.09448133,47.4430463 6.13892578,46.3972038 C4.18337022,45.3513612 2.64863878,43.8166298 1.60279622,41.8610742 C0.556953667,39.9055187 4.4408921e-16,37.9658012 0,32.6166888 L4.4408921e-16,15.3833112 C0,10.0341988 0.556953667,8.09448133 1.60279622,6.13892578 C2.64863878,4.18337022 4.18337022,2.64863878 6.13892578,1.60279622 C8.09448133,0.556953667 10.0341988,4.4408921e-16 15.3833112,0 L32.6166888,1.33226763e-15 C37.9658012,0 39.9055187,0.556953667 41.8610742,1.60279622 C43.8166298,2.64863878 45.3513612,4.18337022 46.3972038,6.13892578 C47.4430463,8.09448133 48,10.0341988 48,15.3833112 L48,32.6166888 C48,32.7918369 47.9986693,32.9777966 47.9960078,33.1745677 L47.9960173,33.1745679 C47.9810785,34.2790416 47.0736153,35.1622842 45.9691416,35.1473454 C45.8647157,35.1459329 45.7605608,35.1363434 45.6576325,35.1186647 Z"
                  id="path-1"
                />
              </mask>
              <image
                xlinkHref={this.props.image}
                x="0"
                y="0"
                width="100%"
                height="100%"
                mask="url(#mask)"
              />

              <g id="Avatar/48">
                <g id="Avatars/48/Square/State/Default">
                  <circle
                    id="Avatars/64/Status/Online"
                    fill={this.props.state}
                    cx="44.25"
                    cy="43.25"
                    r="6.5"
                  />
                </g>
              </g>
            </g>
          </svg>
        </div>
      </AvatarContainer>

      //         );
      //       }}
      //     </TokenConsumer>
      //   </QuantumComponent>
    );
  }
}
