// @framer/ea.quantum/code/ChatBubble.tsx
import * as React from 'react';
import { FrameProperties, PropertyControls, ControlType } from 'framer';

import * as tokens from './tokens';

export interface IChatBubbleProps {
    text: string;
    direction: string;
}

export class ChatBubble extends React.Component<IChatBubbleProps> {
    static defaultProps = {
        text: 'Text that goes into the chat bubble!',
        direction: 'left'
    };

    static propertyControls: PropertyControls = {
        text: {
            title: 'Text',
            type: ControlType.String
        },
        direction: {
            title: 'Direction',
            type: ControlType.SegmentedEnum,
            options: ['left', 'right'],
            optionTitles: ['Left', 'Right']
        }
    };

    render() {
        return (
            <div
                style={{
                    display: 'inline-block',
                    background: this.props.direction === 'right' ? tokens.ColorNeutral._900 : tokens.ColorNeutral._0,
                    borderRadius: tokens.BorderRadiusDefault.medium,
                    borderTopLeftRadius:
                        this.props.direction === 'right'
                            ? tokens.BorderRadiusDefault.medium
                            : tokens.BorderRadiusDefault.smallest,
                    borderBottomRightRadius:
                        this.props.direction === 'right'
                            ? tokens.BorderRadiusDefault.smallest
                            : tokens.BorderRadiusDefault.medium,
                    marginBottom: 4
                }}>
                <p
                    style={{
                        color: this.props.direction === 'right' ? tokens.TextColor.light : tokens.TextColor.dark,
                        margin: 12,
                        lineHeight: tokens.TextSpaceLineHeight.tight,
                        fontFamily: 'BrownPro, Proxima Nova, Helvetica, Arial, sans-serif'
                    }}>
                    {this.props.text}
                </p>
            </div>
        );
    }
}
