// @framer/ea.quantum/code/Dialog.tsx
import * as React from 'react';
import { PropertyControls, ControlType } from 'framer';
import { ButtonGhostIcon } from './Buttons/ButtonGhostIcon';
import styled from 'styled-components';
import * as tokens from './tokens';

const DialogContainer = styled.div`
    height: 100%;
    position: relative;
    background: url(${props => props.image});
    background-color: ${tokens.ColorNeutral._500};
    border-radius: 16px;
    overflow: hidden;
    button.closeButton {
        position: absolute;
        top: 8px;
        right: 8px;
        z-index: 10;
    }
    .children {
        position: absolute;
        top: 40px;
        left: 0;
        right: 0;
    }
`;

// Define type of property
interface Props {
    image: string;
    buttonIcon: any;
}

export class Dialog extends React.Component<Props> {
    // Set default properties
    static defaultProps = {};

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        image: { type: ControlType.Image, title: 'Background Image' }
    };

    render() {
        return (
            <DialogContainer image={this.props.image}>
                <ButtonGhostIcon icon='Close' size='small' iconColor='Light' className='closeButton' />
                <div className='children'>{this.props.children}</div>
            </DialogContainer>
        );
    }
}
