// @framer/ea.quantum/code/GameTile.tsx
import * as React from 'react';
import { PropertyControls, ControlType } from 'framer';
import styled from 'styled-components';
import * as tokens from './tokens';

const GameTileContainer = styled.div`
    height: 100%;
    position: relative;
    background: url(${props => props.image});
    background-position: center center;
    background-size: cover;
    background-color: gray;
    border-radius: 16px;
    transition: transform ${tokens.AnimationTimeDefault.average};

    &:hover {
        transform: scale(1.1);
        transition-timing-function: ${tokens.AnimationEasing.standard};
    }
`;

// Define type of property
interface Props {
    image: string;
}

export class GameTile extends React.Component<Props> {
    // Set default properties
    static defaultProps = {
        width: 114,
        height: 187
    };

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        image: { type: ControlType.Image, title: 'Image' }
    };

    render() {
        return <GameTileContainer image={this.props.image} />;
    }
}
