// @framer/ea.quantum/code/RichObject.tsx
import * as React from 'react';
import { Frame, FrameProperties, PropertyControls, ControlType } from 'framer';
import * as tokens from './tokens';
import * as moment from 'moment';

export interface RichObjectProps extends FrameProperties {
    text: string;
    direction: string;
    iconLeft: any;
    iconRight: any;
    timestamp: number;
}

export class RichObject extends React.Component<Partial<RichObjectProps>> {
    static defaultProps = {
        text: 'Title goes here',
        messages: { timestamp: Date.now() },
        direction: 'left',
        width: 400
    };

    static propertyControls: PropertyControls = {
        text: { title: 'Title', type: ControlType.String },
        iconLeft: {
            type: ControlType.File,
            title: 'Icon L',
            allowedFileTypes: ['svg', 'png', 'jpg', 'gif']
        },
        iconRight: {
            type: ControlType.File,
            title: 'Icon R',
            allowedFileTypes: ['svg', 'png', 'jpg', 'gif']
        },
        direction: {
            title: 'Direction',
            type: ControlType.SegmentedEnum,
            options: ['left', 'right'],
            optionTitles: ['Left', 'Right']
        }
    };

    render() {
        return (
            <div style={{ height: '100%' }}>
                <div
                    style={{
                        display: 'block',
                        position: 'relative',
                        height: '100%',
                        overflow: 'hidden',
                        background: tokens.ColorNeutral._300,
                        borderRadius: tokens.BorderRadiusDefault.medium,
                        borderTopLeftRadius:
                            this.props.direction === 'right'
                                ? tokens.BorderRadiusDefault.medium
                                : tokens.BorderRadiusDefault.smallest,
                        borderBottomRightRadius:
                            this.props.direction === 'right'
                                ? tokens.BorderRadiusDefault.smallest
                                : tokens.BorderRadiusDefault.medium,
                        marginBottom: 4
                    }}>
                    <div
                        style={{
                            position: 'absolute',
                            marginTop: 16,
                            marginBottom: 16,
                            marginLeft: 16,
                            top: 0
                        }}>
                        <div style={{ display: 'flex', alignItems: 'center' }}>
                            <div
                                style={{
                                    width: '30px',
                                    height: '30px',
                                    overflow: 'hidden',
                                    display: 'flex',
                                    alignItems: 'center',
                                    justifyContent: 'center'
                                }}>
                                <img style={{}} src={this.props.iconLeft} />
                            </div>
                            <p
                                style={{
                                    margin: 0,
                                    marginLeft: 8,
                                    display: 'inline-block',
                                    fontFamily: 'BrownPro',
                                    color: 'white',
                                    fontSize: tokens.TextSizeDataDefault.medium
                                }}>
                                {this.props.text}
                            </p>
                        </div>
                    </div>
                    <div
                        style={{
                            width: '30px',
                            height: '30px',
                            overflow: 'hidden',
                            display: 'flex',
                            alignItems: 'center',
                            justifyContent: 'center',
                            position: 'absolute',
                            right: '16px',
                            top: '16px'
                        }}>
                        <img style={{}} src={this.props.iconRight} />
                    </div>
                    <div
                        style={{
                            width: '100%',
                            position: 'absolute',
                            top: '62px',
                            left: 0,
                            margin: 0
                        }}>
                        {this.props.children}
                    </div>
                </div>
                <div style={{ width: '100%' }}>
                    <p
                        style={{
                            color: tokens.TextColor.neutral,
                            fontSize: tokens.TextSizeDataDefault.small,
                            textTransform: 'uppercase',
                            whiteSpace: 'nowrap',
                            overflow: 'hidden',
                            textOverflow: 'ellipsis',
                            textAlign: this.props.direction === 'left' ? 'left' : 'right',
                            fontFamily: 'Apercu Mono Pro, Proxima Nova, Helvetica, Arial, sans-serif',
                            fontWeight: 500
                        }}>
                        {moment(this.props.timestamp).format('LT')}
                    </p>
                </div>
            </div>
        );
    }
}
