// @framer/ea.quantum/code/tokens.ts
/**
 * theme: quantum
 * format: enums.ts
 * platform: web
 * overrides: ["web"]
 * filters: {"breakpoints":[]}
 */

/* Token Enum and Class Declarations */
export enum Breakpoints {
    mobile = 'screen and (min-width: 0px) and (max-width: 416px), screen and (min-width: 417px) and (max-width: 680px), screen and (min-width: 681px) and (max-width: 920px), screen and (min-width: 921px) and (max-width: 1120px)',
    mobileSmall = 'screen and (min-width: 0px) and (max-width: 416px)',
    mobileMedium = 'screen and (min-width: 417px) and (max-width: 680px)',
    mobileLarge = 'screen and (min-width: 681px) and (max-width: 920px)',
    mobileLarger = 'screen and (min-width: 921px) and (max-width: 1120px)',
    desktop = 'screen and (min-width: 1121px) and (max-width: 1320px), screen and (min-width: 1321px) and (max-width: 1520px), screen and (min-width: 1521px) and (max-width: 1720px), screen and (min-width: 1721px)',
    desktopSmall = 'screen and (min-width: 1121px) and (max-width: 1320px)',
    desktopMedium = 'screen and (min-width: 1321px) and (max-width: 1520px)',
    desktopLarge = 'screen and (min-width: 1521px) and (max-width: 1720px)',
    desktopLarger = 'screen and (min-width: 1721px)',
    tv = 'screen and (width: 1920px) and (height: 1080px)'
}
export enum AnimationEasing {
    acceleration = 'cubic-bezier(0.755, 0.05, 0.855, 0.06)',
    accelerationBack = 'cubic-bezier(0.6, -0.28, 0.735, 0.045)',
    deceleration = 'cubic-bezier(0.23, 1, 0.32, 1)',
    decelerationBack = 'cubic-bezier(0.175, 0.885, 0.32, 1.275)',
    linear = 'cubic-bezier(0, 0, 1, 1)',
    standard = 'cubic-bezier(0.86, 0, 0.07, 1)',
    standardBack = 'cubic-bezier(0.68, -0.55, 0.265, 1.55)'
}
export enum AnimationTimeDefault {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeDesktop {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum AnimationTimeDesktopLarge {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum AnimationTimeDesktopLarger {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum AnimationTimeDesktopMedium {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum AnimationTimeDesktopSmall {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum AnimationTimeMobile {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeMobileLarge {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeMobileLarger {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeMobileMedium {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeMobileSmall {
    average = '0.125s',
    crawl = '0.625s',
    fast = '0.1s',
    faster = '0.075s',
    fastest = '0.05s',
    rapid = '0.025s',
    slow = '0.25s',
    slower = '0.375s',
    slowest = '0.5s'
}
export enum AnimationTimeTv {
    average = '0.225s',
    crawl = '1.125s',
    fast = '0.18s',
    faster = '0.135s',
    fastest = '0.09s',
    rapid = '0.045s',
    slow = '0.45s',
    slower = '0.675s',
    slowest = '0.9s'
}
export enum ColorButton {
    ghost = 'rgba(23, 26, 41, 0.6)',
    hollow = 'rgba(23, 26, 41, 0.8)',
    solid = 'rgb(39, 106, 252)'
}
export enum ColorButtonRgb {
    ghost = '23, 26, 41',
    hollow = '23, 26, 41',
    solid = '39, 106, 252'
}
export enum PaddingButtonSquishDefault {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishDesktop {
    large = '24px 48px',
    macro = '36px 72px',
    medium = '18px 36px',
    small = '12px 24px'
}
export enum PaddingButtonSquishDesktopLarge {
    large = '24px 48px',
    macro = '36px 72px',
    medium = '18px 36px',
    small = '12px 24px'
}
export enum PaddingButtonSquishDesktopLarger {
    large = '24px 48px',
    macro = '36px 72px',
    medium = '18px 36px',
    small = '12px 24px'
}
export enum PaddingButtonSquishDesktopMedium {
    large = '24px 48px',
    macro = '36px 72px',
    medium = '18px 36px',
    small = '12px 24px'
}
export enum PaddingButtonSquishDesktopSmall {
    large = '24px 48px',
    macro = '36px 72px',
    medium = '18px 36px',
    small = '12px 24px'
}
export enum PaddingButtonSquishMobile {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishMobileLarge {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishMobileLarger {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishMobileMedium {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishMobileSmall {
    large = '16px 32px',
    macro = '24px 48px',
    medium = '12px 24px',
    small = '8px 16px'
}
export enum PaddingButtonSquishTv {
    large = '32px 64px',
    macro = '48px 96px',
    medium = '24px 48px',
    small = '16px 32px'
}
export enum TextSizeButtonDefault {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonDesktop {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonDesktopLarge {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonDesktopLarger {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonDesktopMedium {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonDesktopSmall {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonMobile {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonMobileLarge {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonMobileLarger {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonMobileMedium {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonMobileSmall {
    large = '20px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeButtonTv {
    large = '32px',
    medium = '24px',
    small = '18px'
}
export enum ColorAlternate {
    primary = 'rgb(131, 104, 253)',
    secondary = 'rgb(255, 109, 77)',
    tertiary = 'rgb(112, 252, 255)'
}
export enum ColorAlternateRgb {
    primary = '131, 104, 253',
    secondary = '255, 109, 77',
    tertiary = '112, 252, 255'
}
export enum ColorBrand {
    primary = 'rgb(39, 106, 252)',
    secondary = 'rgb(255, 0, 82)',
    tertiary = 'rgb(51, 255, 157)'
}
export enum ColorBrandRgb {
    primary = '39, 106, 252',
    secondary = '255, 0, 82',
    tertiary = '51, 255, 157'
}
export enum ColorNeutral {
    _0 = 'rgb(255, 255, 255)',
    _100 = 'rgb(213, 208, 211)',
    _1000 = 'rgb(0, 0, 0)',
    _200 = 'rgb(144, 142, 151)',
    _300 = 'rgb(98, 98, 111)',
    _400 = 'rgb(52, 54, 71)',
    _500 = 'rgb(29, 32, 51)',
    _600 = 'rgb(26, 29, 46)',
    _700 = 'rgb(23, 26, 41)',
    _800 = 'rgb(20, 23, 36)',
    _900 = 'rgb(11, 14, 21)'
}
export enum ColorNeutralRgb {
    _0 = '255, 255, 255',
    _100 = '213, 208, 211',
    _1000 = '0, 0, 0',
    _200 = '144, 142, 151',
    _300 = '98, 98, 111',
    _400 = '52, 54, 71',
    _500 = '29, 32, 51',
    _600 = '26, 29, 46',
    _700 = '23, 26, 41',
    _800 = '20, 23, 36',
    _900 = '11, 14, 21'
}
export enum ColorSystem {
    error = 'rgb(249, 71, 71)',
    success = 'rgb(0, 255, 132)',
    warning = 'rgb(255, 230, 107)'
}
export enum ColorSystemRgb {
    error = '249, 71, 71',
    success = '0, 255, 132',
    warning = '255, 230, 107'
}
export enum ColorGradient {
    primary = 'linear-gradient(-45deg, rgb(255, 0, 82), rgb(255, 109, 77))',
    secondary = 'linear-gradient(-45deg, rgb(39, 106, 252), rgb(131, 104, 253))',
    tertiary = 'linear-gradient(-45deg, rgb(51, 255, 157), rgb(112, 252, 255))'
}
export enum LayoutMarginInlineDefault {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineDesktop {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineDesktopLarge {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineDesktopLarger {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineDesktopMedium {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineDesktopSmall {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineMobile {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineMobileLarge {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineMobileLarger {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineMobileMedium {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineMobileSmall {
    large = '0px 160px 0px 0px',
    larger = '0px 192px 0px 0px',
    largest = '0px 224px 0px 0px',
    macro = '0px 256px 0px 0px',
    medium = '0px 128px 0px 0px',
    micro = '0px 16px 0px 0px',
    small = '0px 96px 0px 0px',
    smaller = '0px 64px 0px 0px',
    smallest = '0px 32px 0px 0px'
}
export enum LayoutMarginInlineTv {
    large = '0px 320px 0px 0px',
    larger = '0px 384px 0px 0px',
    largest = '0px 448px 0px 0px',
    macro = '0px 512px 0px 0px',
    medium = '0px 256px 0px 0px',
    micro = '0px 32px 0px 0px',
    small = '0px 192px 0px 0px',
    smaller = '0px 128px 0px 0px',
    smallest = '0px 64px 0px 0px'
}
export enum LayoutMarginStackedDefault {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedDesktop {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedDesktopLarge {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedDesktopLarger {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedDesktopMedium {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedDesktopSmall {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedMobile {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedMobileLarge {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedMobileLarger {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedMobileMedium {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedMobileSmall {
    large = '0px 0px 160px 0px',
    larger = '0px 0px 192px 0px',
    largest = '0px 0px 224px 0px',
    macro = '0px 0px 256px 0px',
    medium = '0px 0px 128px 0px',
    micro = '0px 0px 16px 0px',
    small = '0px 0px 96px 0px',
    smaller = '0px 0px 64px 0px',
    smallest = '0px 0px 32px 0px'
}
export enum LayoutMarginStackedTv {
    large = '0px 0px 320px 0px',
    larger = '0px 0px 384px 0px',
    largest = '0px 0px 448px 0px',
    macro = '0px 0px 512px 0px',
    medium = '0px 0px 256px 0px',
    micro = '0px 0px 32px 0px',
    small = '0px 0px 192px 0px',
    smaller = '0px 0px 128px 0px',
    smallest = '0px 0px 64px 0px'
}
export enum DimensionLayoutDefault {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutDesktop {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutDesktopLarge {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutDesktopLarger {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutDesktopMedium {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutDesktopSmall {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutMobile {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutMobileLarge {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutMobileLarger {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutMobileMedium {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutMobileSmall {
    large = '160px',
    larger = '192px',
    largest = '224px',
    macro = '256px',
    medium = '128px',
    micro = '16px',
    none = '0px',
    small = '96px',
    smaller = '64px',
    smallest = '32px'
}
export enum DimensionLayoutTv {
    large = '320px',
    larger = '384px',
    largest = '448px',
    macro = '512px',
    medium = '256px',
    micro = '32px',
    none = '0px',
    small = '192px',
    smaller = '128px',
    smallest = '64px'
}
export enum MediaQueryDesktopLarge {
    maxWidth = '1720px',
    minWidth = '1521px'
}
export enum MediaQueryDesktopLarger {
    minWidth = '1721px'
}
export enum MediaQueryDesktopMedium {
    maxWidth = '1520px',
    minWidth = '1321px'
}
export enum MediaQueryDesktopSmall {
    maxWidth = '1320px',
    minWidth = '1121px'
}
export enum MediaQueryMobileLarge {
    maxWidth = '920px',
    minWidth = '681px'
}
export enum MediaQueryMobileLarger {
    maxWidth = '1120px',
    minWidth = '921px'
}
export enum MediaQueryMobileMedium {
    maxWidth = '680px',
    minWidth = '417px'
}
export enum MediaQueryMobileSmall {
    maxWidth = '416px',
    minWidth = '0px'
}
export enum MediaQueryTv {
    height = '1080px',
    width = '1920px'
}
export enum TextSizeNavigationDefault {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationDesktop {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationDesktopLarge {
    large = '24px',
    medium = '18px',
    small = '16px'
}
export enum TextSizeNavigationDesktopLarger {
    large = '24px',
    medium = '18px',
    small = '16px'
}
export enum TextSizeNavigationDesktopMedium {
    large = '24px',
    medium = '18px',
    small = '16px'
}
export enum TextSizeNavigationDesktopSmall {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationMobile {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationMobileLarge {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationMobileLarger {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationMobileMedium {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationMobileSmall {
    large = '20px',
    medium = '16px',
    small = '14px'
}
export enum TextSizeNavigationTv {
    large = '32px',
    medium = '24px',
    small = '18px'
}
export enum EffectsRadiationAlpha {
    large = 0.2,
    medium = 0.3,
    small = 0.4
}
export enum EffectsRadiationBrandPrimary {
    large = '0px 0px 80px 0 rgba(39, 106, 252, 0.2)',
    medium = '0px 0px 30px 0 rgba(39, 106, 252, 0.3)',
    small = '0px 0px 10px 0 rgba(39, 106, 252, 0.4)'
}
export enum EffectsRadiationBrandSecondary {
    large = '0px 0px 80px 0 rgba(255, 0, 82, 0.2)',
    medium = '0px 0px 30px 0 rgba(255, 0, 82, 0.3)',
    small = '0px 0px 10px 0 rgba(255, 0, 82, 0.4)'
}
export enum EffectsRadiationBrandTertiary {
    large = '0px 0px 80px 0 rgba(51, 255, 157, 0.2)',
    medium = '0px 0px 30px 0 rgba(51, 255, 157, 0.3)',
    small = '0px 0px 10px 0 rgba(51, 255, 157, 0.4)'
}
export enum EffectsRadiationRadius {
    large = '80px',
    medium = '30px',
    small = '10px'
}
export enum EffectsRadiationSystemError {
    large = '0px 0px 80px 0 rgba(249, 71, 71, 0.2)',
    medium = '0px 0px 30px 0 rgba(249, 71, 71, 0.3)',
    small = '0px 0px 10px 0 rgba(249, 71, 71, 0.4)'
}
export enum EffectsRadiationSystemSuccess {
    large = '0px 0px 80px 0 rgba(0, 255, 132, 0.2)',
    medium = '0px 0px 30px 0 rgba(0, 255, 132, 0.3)',
    small = '0px 0px 10px 0 rgba(0, 255, 132, 0.4)'
}
export enum EffectsRadiationSystemWarning {
    large = '0px 0px 80px 0 rgba(255, 230, 107, 0.2)',
    medium = '0px 0px 30px 0 rgba(255, 230, 107, 0.3)',
    small = '0px 0px 10px 0 rgba(255, 230, 107, 0.4)'
}
export enum FillAlpha {
    high = 0.1,
    low = 0.9,
    medium = 0.5
}
export enum BorderRadiusDefault {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusDesktop {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusDesktopLarge {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusDesktopLarger {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusDesktopMedium {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusDesktopSmall {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusMobile {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusMobileLarge {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusMobileLarger {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusMobileMedium {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusMobileSmall {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '64px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum BorderRadiusTv {
    large = '48px',
    larger = '64px',
    largest = '96px',
    macro = '128px',
    medium = '32px',
    micro = '4px',
    none = '0px',
    small = '24px',
    smaller = '16px',
    smallest = '8px'
}
export enum StrokeDashDefault {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashDesktop {
    gap = '18px',
    length = '12px'
}
export enum StrokeDashDesktopLarge {
    gap = '18px',
    length = '12px'
}
export enum StrokeDashDesktopLarger {
    gap = '18px',
    length = '12px'
}
export enum StrokeDashDesktopMedium {
    gap = '18px',
    length = '12px'
}
export enum StrokeDashDesktopSmall {
    gap = '18px',
    length = '12px'
}
export enum StrokeDashMobile {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashMobileLarge {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashMobileLarger {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashMobileMedium {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashMobileSmall {
    gap = '12px',
    length = '8px'
}
export enum StrokeDashTv {
    gap = '24px',
    length = '16px'
}
export enum StrokeEnd {
    cap = 'round'
}
export enum StrokeWidthDefault {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthDesktop {
    large = '12px',
    medium = '6px',
    small = '3px'
}
export enum StrokeWidthDesktopLarge {
    large = '12px',
    medium = '6px',
    small = '3px'
}
export enum StrokeWidthDesktopLarger {
    large = '12px',
    medium = '6px',
    small = '3px'
}
export enum StrokeWidthDesktopMedium {
    large = '12px',
    medium = '6px',
    small = '3px'
}
export enum StrokeWidthDesktopSmall {
    large = '12px',
    medium = '6px',
    small = '3px'
}
export enum StrokeWidthMobile {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthMobileLarge {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthMobileLarger {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthMobileMedium {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthMobileSmall {
    large = '8px',
    medium = '4px',
    small = '2px'
}
export enum StrokeWidthTv {
    large = '16px',
    medium = '8px',
    small = '4px'
}
export enum SpaceMarginInlineDefault {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineDesktop {
    large = '0px 36px 0px 0px',
    larger = '0px 48px 0px 0px',
    largest = '0px 72px 0px 0px',
    macro = '0px 120px 0px 0px',
    medium = '0px 24px 0px 0px',
    micro = '0px 3px 0px 0px',
    small = '0px 18px 0px 0px',
    smaller = '0px 12px 0px 0px',
    smallest = '0px 6px 0px 0px'
}
export enum SpaceMarginInlineDesktopLarge {
    large = '0px 36px 0px 0px',
    larger = '0px 48px 0px 0px',
    largest = '0px 72px 0px 0px',
    macro = '0px 120px 0px 0px',
    medium = '0px 24px 0px 0px',
    micro = '0px 3px 0px 0px',
    small = '0px 18px 0px 0px',
    smaller = '0px 12px 0px 0px',
    smallest = '0px 6px 0px 0px'
}
export enum SpaceMarginInlineDesktopLarger {
    large = '0px 36px 0px 0px',
    larger = '0px 48px 0px 0px',
    largest = '0px 72px 0px 0px',
    macro = '0px 120px 0px 0px',
    medium = '0px 24px 0px 0px',
    micro = '0px 3px 0px 0px',
    small = '0px 18px 0px 0px',
    smaller = '0px 12px 0px 0px',
    smallest = '0px 6px 0px 0px'
}
export enum SpaceMarginInlineDesktopMedium {
    large = '0px 36px 0px 0px',
    larger = '0px 48px 0px 0px',
    largest = '0px 72px 0px 0px',
    macro = '0px 120px 0px 0px',
    medium = '0px 24px 0px 0px',
    micro = '0px 3px 0px 0px',
    small = '0px 18px 0px 0px',
    smaller = '0px 12px 0px 0px',
    smallest = '0px 6px 0px 0px'
}
export enum SpaceMarginInlineDesktopSmall {
    large = '0px 36px 0px 0px',
    larger = '0px 48px 0px 0px',
    largest = '0px 72px 0px 0px',
    macro = '0px 120px 0px 0px',
    medium = '0px 24px 0px 0px',
    micro = '0px 3px 0px 0px',
    small = '0px 18px 0px 0px',
    smaller = '0px 12px 0px 0px',
    smallest = '0px 6px 0px 0px'
}
export enum SpaceMarginInlineMobile {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineMobileLarge {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineMobileLarger {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineMobileMedium {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineMobileSmall {
    large = '0px 24px 0px 0px',
    larger = '0px 32px 0px 0px',
    largest = '0px 48px 0px 0px',
    macro = '0px 80px 0px 0px',
    medium = '0px 16px 0px 0px',
    micro = '0px 2px 0px 0px',
    small = '0px 12px 0px 0px',
    smaller = '0px 8px 0px 0px',
    smallest = '0px 4px 0px 0px'
}
export enum SpaceMarginInlineTv {
    large = '0px 48px 0px 0px',
    larger = '0px 64px 0px 0px',
    largest = '0px 96px 0px 0px',
    macro = '0px 160px 0px 0px',
    medium = '0px 32px 0px 0px',
    micro = '0px 4px 0px 0px',
    small = '0px 24px 0px 0px',
    smaller = '0px 16px 0px 0px',
    smallest = '0px 8px 0px 0px'
}
export enum SpaceMarginStackedDefault {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedDesktop {
    large = '0px 0px 36px 0px',
    larger = '0px 0px 48px 0px',
    largest = '0px 0px 72px 0px',
    macro = '0px 0px 120px 0px',
    medium = '0px 0px 24px 0px',
    micro = '0px 0px 3px 0px',
    small = '0px 0px 18px 0px',
    smaller = '0px 0px 12px 0px',
    smallest = '0px 0px 6px 0px'
}
export enum SpaceMarginStackedDesktopLarge {
    large = '0px 0px 36px 0px',
    larger = '0px 0px 48px 0px',
    largest = '0px 0px 72px 0px',
    macro = '0px 0px 120px 0px',
    medium = '0px 0px 24px 0px',
    micro = '0px 0px 3px 0px',
    small = '0px 0px 18px 0px',
    smaller = '0px 0px 12px 0px',
    smallest = '0px 0px 6px 0px'
}
export enum SpaceMarginStackedDesktopLarger {
    large = '0px 0px 36px 0px',
    larger = '0px 0px 48px 0px',
    largest = '0px 0px 72px 0px',
    macro = '0px 0px 120px 0px',
    medium = '0px 0px 24px 0px',
    micro = '0px 0px 3px 0px',
    small = '0px 0px 18px 0px',
    smaller = '0px 0px 12px 0px',
    smallest = '0px 0px 6px 0px'
}
export enum SpaceMarginStackedDesktopMedium {
    large = '0px 0px 36px 0px',
    larger = '0px 0px 48px 0px',
    largest = '0px 0px 72px 0px',
    macro = '0px 0px 120px 0px',
    medium = '0px 0px 24px 0px',
    micro = '0px 0px 3px 0px',
    small = '0px 0px 18px 0px',
    smaller = '0px 0px 12px 0px',
    smallest = '0px 0px 6px 0px'
}
export enum SpaceMarginStackedDesktopSmall {
    large = '0px 0px 36px 0px',
    larger = '0px 0px 48px 0px',
    largest = '0px 0px 72px 0px',
    macro = '0px 0px 120px 0px',
    medium = '0px 0px 24px 0px',
    micro = '0px 0px 3px 0px',
    small = '0px 0px 18px 0px',
    smaller = '0px 0px 12px 0px',
    smallest = '0px 0px 6px 0px'
}
export enum SpaceMarginStackedMobile {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedMobileLarge {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedMobileLarger {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedMobileMedium {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedMobileSmall {
    large = '0px 0px 24px 0px',
    larger = '0px 0px 32px 0px',
    largest = '0px 0px 48px 0px',
    macro = '0px 0px 80px 0px',
    medium = '0px 0px 16px 0px',
    micro = '0px 0px 2px 0px',
    small = '0px 0px 12px 0px',
    smaller = '0px 0px 8px 0px',
    smallest = '0px 0px 4px 0px'
}
export enum SpaceMarginStackedTv {
    large = '0px 0px 48px 0px',
    larger = '0px 0px 64px 0px',
    largest = '0px 0px 96px 0px',
    macro = '0px 0px 160px 0px',
    medium = '0px 0px 32px 0px',
    micro = '0px 0px 4px 0px',
    small = '0px 0px 24px 0px',
    smaller = '0px 0px 16px 0px',
    smallest = '0px 0px 8px 0px'
}
export enum DimensionSpaceDefault {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceDesktop {
    large = '36px',
    larger = '48px',
    largest = '72px',
    macro = '120px',
    medium = '24px',
    micro = '3px',
    none = '0px',
    small = '18px',
    smaller = '12px',
    smallest = '6px'
}
export enum DimensionSpaceDesktopLarge {
    large = '36px',
    larger = '48px',
    largest = '72px',
    macro = '120px',
    medium = '24px',
    micro = '3px',
    none = '0px',
    small = '18px',
    smaller = '12px',
    smallest = '6px'
}
export enum DimensionSpaceDesktopLarger {
    large = '36px',
    larger = '48px',
    largest = '72px',
    macro = '120px',
    medium = '24px',
    micro = '3px',
    none = '0px',
    small = '18px',
    smaller = '12px',
    smallest = '6px'
}
export enum DimensionSpaceDesktopMedium {
    large = '36px',
    larger = '48px',
    largest = '72px',
    macro = '120px',
    medium = '24px',
    micro = '3px',
    none = '0px',
    small = '18px',
    smaller = '12px',
    smallest = '6px'
}
export enum DimensionSpaceDesktopSmall {
    large = '36px',
    larger = '48px',
    largest = '72px',
    macro = '120px',
    medium = '24px',
    micro = '3px',
    none = '0px',
    small = '18px',
    smaller = '12px',
    smallest = '6px'
}
export enum DimensionSpaceMobile {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceMobileLarge {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceMobileLarger {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceMobileMedium {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceMobileSmall {
    large = '24px',
    larger = '32px',
    largest = '48px',
    macro = '80px',
    medium = '16px',
    micro = '2px',
    none = '0px',
    small = '12px',
    smaller = '8px',
    smallest = '4px'
}
export enum DimensionSpaceTv {
    large = '48px',
    larger = '64px',
    largest = '96px',
    macro = '160px',
    medium = '32px',
    micro = '4px',
    none = '0px',
    small = '24px',
    smaller = '16px',
    smallest = '8px'
}
export enum SpacePaddingSquashDefault {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashDesktop {
    large = '36px 24px',
    larger = '48px 36px',
    largest = '72px 48px',
    macro = '120px 72px',
    medium = '24px 18px',
    micro = '3px 0px',
    small = '18px 12px',
    smaller = '12px 6px',
    smallest = '6px 3px'
}
export enum SpacePaddingSquashDesktopLarge {
    large = '36px 24px',
    larger = '48px 36px',
    largest = '72px 48px',
    macro = '120px 72px',
    medium = '24px 18px',
    micro = '3px 0px',
    small = '18px 12px',
    smaller = '12px 6px',
    smallest = '6px 3px'
}
export enum SpacePaddingSquashDesktopLarger {
    large = '36px 24px',
    larger = '48px 36px',
    largest = '72px 48px',
    macro = '120px 72px',
    medium = '24px 18px',
    micro = '3px 0px',
    small = '18px 12px',
    smaller = '12px 6px',
    smallest = '6px 3px'
}
export enum SpacePaddingSquashDesktopMedium {
    large = '36px 24px',
    larger = '48px 36px',
    largest = '72px 48px',
    macro = '120px 72px',
    medium = '24px 18px',
    micro = '3px 0px',
    small = '18px 12px',
    smaller = '12px 6px',
    smallest = '6px 3px'
}
export enum SpacePaddingSquashDesktopSmall {
    large = '36px 24px',
    larger = '48px 36px',
    largest = '72px 48px',
    macro = '120px 72px',
    medium = '24px 18px',
    micro = '3px 0px',
    small = '18px 12px',
    smaller = '12px 6px',
    smallest = '6px 3px'
}
export enum SpacePaddingSquashMobile {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashMobileLarge {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashMobileLarger {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashMobileMedium {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashMobileSmall {
    large = '24px 16px',
    larger = '32px 24px',
    largest = '48px 32px',
    macro = '80px 48px',
    medium = '16px 12px',
    micro = '2px 0px',
    small = '12px 8px',
    smaller = '8px 4px',
    smallest = '4px 2px'
}
export enum SpacePaddingSquashTv {
    large = '48px 32px',
    larger = '64px 48px',
    largest = '96px 64px',
    macro = '160px 96px',
    medium = '32px 24px',
    micro = '4px 0px',
    small = '24px 16px',
    smaller = '16px 8px',
    smallest = '8px 4px'
}
export enum SpacePaddingSquishDefault {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishDesktop {
    large = '24px 36px',
    larger = '36px 48px',
    largest = '48px 72px',
    macro = '72px 120px',
    medium = '18px 24px',
    micro = '0px 3px',
    small = '12px 18px',
    smaller = '6px 12px',
    smallest = '3px 6px'
}
export enum SpacePaddingSquishDesktopLarge {
    large = '24px 36px',
    larger = '36px 48px',
    largest = '48px 72px',
    macro = '72px 120px',
    medium = '18px 24px',
    micro = '0px 3px',
    small = '12px 18px',
    smaller = '6px 12px',
    smallest = '3px 6px'
}
export enum SpacePaddingSquishDesktopLarger {
    large = '24px 36px',
    larger = '36px 48px',
    largest = '48px 72px',
    macro = '72px 120px',
    medium = '18px 24px',
    micro = '0px 3px',
    small = '12px 18px',
    smaller = '6px 12px',
    smallest = '3px 6px'
}
export enum SpacePaddingSquishDesktopMedium {
    large = '24px 36px',
    larger = '36px 48px',
    largest = '48px 72px',
    macro = '72px 120px',
    medium = '18px 24px',
    micro = '0px 3px',
    small = '12px 18px',
    smaller = '6px 12px',
    smallest = '3px 6px'
}
export enum SpacePaddingSquishDesktopSmall {
    large = '24px 36px',
    larger = '36px 48px',
    largest = '48px 72px',
    macro = '72px 120px',
    medium = '18px 24px',
    micro = '0px 3px',
    small = '12px 18px',
    smaller = '6px 12px',
    smallest = '3px 6px'
}
export enum SpacePaddingSquishMobile {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishMobileLarge {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishMobileLarger {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishMobileMedium {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishMobileSmall {
    large = '16px 24px',
    larger = '24px 32px',
    largest = '32px 48px',
    macro = '48px 80px',
    medium = '12px 16px',
    micro = '0px 2px',
    small = '8px 12px',
    smaller = '4px 8px',
    smallest = '2px 4px'
}
export enum SpacePaddingSquishTv {
    large = '32px 48px',
    larger = '48px 64px',
    largest = '64px 96px',
    macro = '96px 160px',
    medium = '24px 32px',
    micro = '0px 4px',
    small = '16px 24px',
    smaller = '8px 16px',
    smallest = '4px 8px'
}
export enum TextColor {
    attention = 'rgb(255, 0, 82)',
    dark = 'rgb(20, 23, 36)',
    error = 'rgb(249, 71, 71)',
    light = 'rgb(255, 255, 255)',
    neutral = 'rgb(144, 142, 151)',
    success = 'rgb(0, 255, 132)',
    warning = 'rgb(255, 230, 107)'
}
export enum TextColorRgb {
    attention = '255, 0, 82',
    dark = '20, 23, 36',
    error = '249, 71, 71',
    light = '255, 255, 255',
    neutral = '144, 142, 151',
    success = '0, 255, 132',
    warning = '255, 230, 107'
}
export enum TextFamily {
    data = '"Aperçu Pro Mono", monospace',
    primary = '"LL Brown", "Proxima Nova", Arial, sans-serif',
    secondary = '"Proxima Nova", Arial, sans-serif'
}
export enum TextSpaceLetterSpacing {
    allCaps = '0.1px',
    dataLoose = '0.05px',
    dataModerate = '0px',
    display = '-0.02px',
    heading = '0.02px',
    paragraph = '0.03px'
}
export enum TextSpaceLineHeight {
    moderate = 1.618,
    tight = 1.309,
    tighter = 1.129
}
export enum TextSizeOption {
    _100 = '12px',
    _1000 = '42px',
    _1100 = '48px',
    _1200 = '54px',
    _1300 = '60px',
    _1400 = '68px',
    _1500 = '76px',
    _1600 = '84px',
    _1700 = '92px',
    _200 = '14px',
    _300 = '16px',
    _400 = '18px',
    _500 = '20px',
    _600 = '24px',
    _700 = '28px',
    _800 = '32px',
    _900 = '36px'
}
export enum TextSizeDataDefault {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataDesktop {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataDesktopLarge {
    large = '24px',
    larger = '36px',
    medium = '18px',
    small = '12px'
}
export enum TextSizeDataDesktopLarger {
    large = '24px',
    larger = '36px',
    medium = '18px',
    small = '12px'
}
export enum TextSizeDataDesktopMedium {
    large = '24px',
    larger = '36px',
    medium = '18px',
    small = '12px'
}
export enum TextSizeDataDesktopSmall {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataMobile {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataMobileLarge {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataMobileLarger {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataMobileMedium {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataMobileSmall {
    large = '20px',
    larger = '32px',
    medium = '16px',
    small = '12px'
}
export enum TextSizeDataTv {
    large = '32px',
    larger = '48px',
    medium = '24px',
    small = '18px'
}
export enum TextSizeDisplayDefault {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayDesktop {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayDesktopLarge {
    medium = '76px',
    small = '48px'
}
export enum TextSizeDisplayDesktopLarger {
    medium = '76px',
    small = '48px'
}
export enum TextSizeDisplayDesktopMedium {
    medium = '76px',
    small = '48px'
}
export enum TextSizeDisplayDesktopSmall {
    medium = '60px',
    small = '42px'
}
export enum TextSizeDisplayMobile {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayMobileLarge {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayMobileLarger {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayMobileMedium {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayMobileSmall {
    medium = '42px',
    small = '36px'
}
export enum TextSizeDisplayTv {
    medium = '92px',
    small = '60px'
}
export enum TextSizeHeadingDefault {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingDesktop {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingDesktopLarge {
    large = '32px',
    medium = '24px',
    small = '18px',
    smaller = '14px'
}
export enum TextSizeHeadingDesktopLarger {
    large = '32px',
    medium = '24px',
    small = '18px',
    smaller = '14px'
}
export enum TextSizeHeadingDesktopMedium {
    large = '32px',
    medium = '24px',
    small = '18px',
    smaller = '14px'
}
export enum TextSizeHeadingDesktopSmall {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingMobile {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingMobileLarge {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingMobileLarger {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingMobileMedium {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingMobileSmall {
    large = '28px',
    medium = '20px',
    small = '16px',
    smaller = '12px'
}
export enum TextSizeHeadingTv {
    large = '42px',
    medium = '32px',
    small = '24px',
    smaller = '18px'
}
export enum TextSizeLabelDefault {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelDesktop {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelDesktopLarge {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelDesktopLarger {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelDesktopMedium {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelDesktopSmall {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelMobile {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelMobileLarge {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelMobileLarger {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelMobileMedium {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelMobileSmall {
    medium = '16px',
    small = '12px'
}
export enum TextSizeLabelTv {
    medium = '24px',
    small = '18px'
}
export enum TextSizeParagraphDefault {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphDesktop {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphDesktopLarge {
    large = '24px',
    medium = '18px',
    small = '16px',
    smaller = '14px'
}
export enum TextSizeParagraphDesktopLarger {
    large = '24px',
    medium = '18px',
    small = '16px',
    smaller = '14px'
}
export enum TextSizeParagraphDesktopMedium {
    large = '24px',
    medium = '18px',
    small = '16px',
    smaller = '14px'
}
export enum TextSizeParagraphDesktopSmall {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphMobile {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphMobileLarge {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphMobileLarger {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphMobileMedium {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphMobileSmall {
    large = '20px',
    medium = '16px',
    small = '14px',
    smaller = '12px'
}
export enum TextSizeParagraphTv {
    large = '32px',
    medium = '24px',
    small = '20px',
    smaller = '18px'
}
