// @framer/ea.articulate/code/Transitions.ts
import {FrameProperties} from 'framer';

export interface ITransition {
    frameA: ITransitionFrame;
    frameB: ITransitionFrame;
}

export interface ITransitionFrame {
    show: Partial<FrameProperties>;
    hide: Partial<FrameProperties>;
    options: ITransitionOptions;
}

export interface ITransitionOptions {
    time: number;
    easing?: string;
    delay?: number;
}

export interface ITransitions {
    [key: string]: Function;
}

export const Transitions: ITransitions = {
    dissolve: (routerProps): ITransition => {
        const options: ITransitionOptions = {
            time: .4
        }

        return {
            frameA: {
                show: {
                    opacity: 1
                },
                hide: {
                    opacity: 0
                },
                options
            },
            frameB: {
                show: {
                    opacity: 1
                },
                hide: {
                    opacity: 0
                },
                options
            }
        }
    },

    depth: (routerProps): ITransition => {
        const options: ITransitionOptions = {
            time: .4
        }

        return {
            frameA: {
                show: {
                    scale: 1,
                    opacity: 1
                },
                hide: {
                    scale: 1.2,
                    opacity: 0
                },
                options
            },
            frameB: {
                show: {
                    scale: 1,
                    opacity: 1
                },
                hide: {
                    scale: .8,
                    opacity: 0
                },
                options
            }
        }
    },

    modal: (routerProps): ITransition => {
        const options: ITransitionOptions = {
            time: .4
        }

        return {
            frameA: {
                show: {
                    top: 0,
                    z: 0,
                    opacity: 1
                },
                hide: {
                    top: 0,
                    z: -50,
                    opacity: 0
                },
                options
            },
            frameB: {
                show: {
                    top: 0,
                    z: 0
                },
                hide: {
                    top: routerProps.height,
                    z: 0
                },
                options
            }
        }
    },

    slideRight: (routerProps): ITransition => {
        const options: ITransitionOptions = {
            time: .4
        }

        return {
            frameA: {
                show: {
                    left: 0
                },
                hide: {
                    left: -routerProps.width
                },
                options
            },
            frameB: {
                show: {
                    left: 0
                },
                hide: {
                    left: routerProps.width
                },
                options
            }
        }
    },

    slideLeft: (routerProps): ITransition => {
        const options: ITransitionOptions = {
            time: .4
        }

        return {
            frameA: {
                show: {
                    left: 0
                },
                hide: {
                    left: routerProps.width
                },
                options
            },
            frameB: {
                show: {
                    left: 0
                },
                hide: {
                    left: -routerProps.width
                },
                options
            }
        }
    }
};
